library/core/src/macros/mod.rs

```diff
@@ -1295,6 +1295,12 @@ pub(crate) mod builtin {
     /// first macro invocation leading up to the invocation of the `file!`
     /// macro.
     ///
+    /// The file name is derived from the source path passed to the Rust compiler and any path operations
+    /// performed on top of that (e.g. `#[path = "<child>"]` is joined to the current `file!`),
+    /// modified by any flags passed to the Rust compiler (e.g. `--remap-path-prefix`).
+    /// If the source path is relative,
+    /// the initial base directory will be the working directory of the Rust compiler.
```

Comment:
I think this needs word-wrapping.

Reply:
I was using [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).  If that goes against the existing style, I can change it.

Reply:
No objections on my part. To the best of my knowledge we don't reliably *have* a clear comment style, because we don't have comment wrapping turned on in our `rustfmt` settings.